uqcsbot/join.py

```diff
@@ -0,0 +1,103 @@
+import discord
+from discord.ext import commands
+
+from uqcsbot.bot import UQCSBot
+from uqcsbot.models import Channel
+
+JOINED_PERMISSIONS = discord.Permissions(read_messages=True)
+SERVER_ID = 813324385179271168
+# Testing Server
+# SERVER_ID = 836589565237264415
+
+EMOJIS = {"academic-advice": "📖", "adulting": "🧑", "banter": "💬", "bot-testing" : "🤖",
+                "contests" : "⚔️", "covid" : "⚛️", "creative" : "🎨", "events" : "🗓️", "food" : "🍔",
+                "games" : "🎮", "general" : "⚪", "hackathons" : "🍕", "hardware" : "💻", "jobs-bulletin" : "📌",
+                "jobs-discussion" : "🔈", "lgbtqia" : "🏳️‍🌈", "media" : "📺", "memes" : "🎭", "politics" : "📮",
+                "projects" : "🔨", "uqic-sport" : "🏅", "yelling" : "🗣️"}
+
+class Join(commands.Cog):
+
+    def __init__(self, bot: UQCSBot):
+        self.bot = bot
+        self.message_id = None
+
+    def _channel_query(self, channel: str):
+        db_session = self.bot.create_db_session()
+        channel_query = db_session.query(Channel).filter(Channel.name == channel,
+                                                         Channel.joinable == True).one_or_none()
+        db_session.close()
+        return channel_query
+
+    def get_key(self, map, value):
```

Comment:
Given the key gets indexed more often, I thought it would be okay to leave the map in the current order. Although this is definitely a method that can be replaced with a one-liner